### PR TITLE
Update EIP-3540 and EIP-3670 Tests to Cancun

### PR DESCRIPTION
**Pull request by hugo-dc**

### CODE REVIEW

**Comment by winsvega** [comment]
no, first letes decouple it from normal tests
https://github.com/ethereum/tests/pull/1192

**Comment by winsvega** [comment]
this might be not needed now. as the tests are in separate folder and don't interfere with Hive

**Comment by hugo-dc** [comment]
I have updated the PR to place the tests in the correct directory. Leaving it as Cancun as it is how Evmone is currently expecting it. Also this PR updates the tests due to new logic of how nonce is incremented in some invalid EOF code.

File: src/EIPTestsFiller/StateTests/stEOF/stEIP3540/CREATE2_EOF1_FromEOFFiller.yml [hunk 1]
```diff
@@ -30,14 +30,10 @@ CREATE2_EOF1_FromEOF:
...
 
-      #- ':yul { mstore8(0, 0xef) return(0, 1) }'
-      - ':label deploying_0xef :raw 0x60ef60005360016000f3'
-      #- ':yul { mstore8(0, 0xef) return(0, 2) }'
-      - ':label deploying_0xef00 :raw 0x60ef60005360026000f3'
-      #- ':yul { mstore8(0, 0xef) return(0, 3) }'
-      - ':label deploying_0xef0000 :raw 0x60ef60005360036000f3'
-      #- ':yul { mstore8(0, 0xef) return(0, 32) }'
-      - ':label deploying_0xef00..00 :raw 0x60ef60005360206000f3'
+      - ':label deploying_0xef :yul-eof { mstore8(0, 0xef) return(0, 1) }'
```
Comment (winsvega) [inline src/EIPTestsFiller/StateTests/stEOF/stEIP3540/CREATE2_EOF1_FromEOFFiller.yml:33]: what is yul-eof let me add it to the config

File: src/EIPTestsFiller/StateTests/stEOF/stEIP3540/CREATE2_EOF1_FromEOFFiller.yml [hunk 2]
```diff
@@ -80,7 +76,7 @@ CREATE2_EOF1_FromEOF:
     - indexes:
        data: ':label deploying_0xef'
       network:
-        - 'Shanghai'
+        - 'Cancun'
```
Comment (winsvega) [inline src/EIPTestsFiller/StateTests/stEOF/stEIP3540/CREATE2_EOF1_FromEOFFiller.yml]: `>=Cancun` so the test get generated a vector after Cancun too

**Comment by winsvega** [comment]
Draft?

File: src/EIPTestsFiller/StateTests/stEOF/stEIP3540/CREATE2_EOF1_FromEOFFiller.yml (hunk 1 above)
Comment (hugo-dc) [reply src/EIPTestsFiller/StateTests/stEOF/stEIP3540/CREATE2_EOF1_FromEOFFiller.yml:33]: Here is the yul-eof script: https://github.com/hugo-dc/retesteth-config/blob/eof/default/yul-eof.sh
It's using solidity from this branch: https://github.com/ethereum/solidity/tree/eof-functions

File: src/EIPTestsFiller/StateTests/stEOF/stEIP3540/CREATE2_EOF1_FromEOFFiller.yml (hunk 2 above)
Comment (hugo-dc) [reply src/EIPTestsFiller/StateTests/stEOF/stEIP3540/CREATE2_EOF1_FromEOFFiller.yml]: Changed.

**Comment by winsvega** [comment]
please rebase on develop

**Review by winsvega — APPROVED** [review]
please rebase on develop

**Pull request merged**